### ISSUE TITLE
star unpacking triggers false positive type error

### ISSUE THREAD

**Issue by asottile** [python/mypy#4979]
### code

```python
import os.path


def f(filename: str) -> str:
    filename, _ = os.path.splitext(filename)
    pkgname, *_ = filename.split('_')
    return pkgname


print(f('foo.json'))
print(f('foo_bar.json'))
```

### output (execution)

```console
$ python test.py
foo
foo
```

### mypy output

```diff
--- test.py	2018-04-26 15:43:43.000000000 -0700
+++ test2.py	2018-04-26 15:44:41.000000000 -0700
@@ -4,6 +4,8 @@
 def f(filename: str) -> str:
     filename, _ = os.path.splitext(filename)
     pkgname, *_ = filename.split('_')
+    reveal_type(filename)
+    reveal_type(pkgname)
     return pkgname
```

```console
$ mypy test2.py
test2.py:6: error: Incompatible types in assignment (expression has type "List[str]", variable has type "str")
test2.py:7: error: Revealed type is 'builtins.str'
test2.py:8: error: Revealed type is 'builtins.str*'
```

### version information

```
$ mypy --version
mypy 0.590
$ python --version --version
Python 3.6.5 (default, Mar 30 2018, 06:41:53) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

### does it pass on master

yes!  I suspect this change: https://github.com/python/mypy/commit/7b257c8de609ebe2552620c4809d7ef97e17d492

**Comment by asottile** [comment]
mostly wanted to document this so others can see it -- thanks for the fix :)

**Comment by asottile** [comment]
A workaround in my case is to just use `str.partition`
